src/com/ms/silverking/cloud/dht/management/MetaUtil.java

```diff
@@ -42,310 +42,316 @@
+  }
+
+  private void displayDHTHostGroupServers(Iterable<String> hostGroups, FilterOption filterOption, String workersFile,
+      String exclusionsFile) throws KeeperException, IOException {
+    Set<String> allowedHostAddresses;
+    Set<String> workers;
+
+    workers = new HashSet<>();
+    allowedHostAddresses = readHostGroupTable().getHostAddresses(hostGroups);
+    for (String dhtServer : getDHTServersFromZk()) {
+      if (allowedHostAddresses.contains(dhtServer)) {
+        if (filterOption == FilterOption.NoFiltering || !exclusions.contains(dhtServer)) {
+          displayForFiltering(dhtServer);
+          workers.add(dhtServer);
         }
+      } else {
+        //System.out.println("Ignoring:\t"+ dhtServer);
+      }
     }
-    
-    private void displayDHTHostGroupServers(Iterable<String> hostGroups, FilterOption filterOption, 
-            String workersFile, String exclusionsFile) throws KeeperException, IOException {
-        Set<String>     allowedHostAddresses;
-        Set<String>     workers;
-        
-        workers = new HashSet<>();
-        allowedHostAddresses = readHostGroupTable().getHostAddresses(hostGroups);
-        for(String dhtServer : getDHTServersFromZk()) {
-            if (allowedHostAddresses.contains(dhtServer)) {
-                if (filterOption == FilterOption.NoFiltering || !exclusions.contains(dhtServer)) {
-                    displayForFiltering(dhtServer);
-                    workers.add(dhtServer);
-                }
-            } else {
-                //System.out.println("Ignoring:\t"+ dhtServer);
-            }
-        }
-        
-        if (exclusionsFile != null) {
-            IOUtil.writeAsLines(new File(exclusionsFile), exclusions.getServers());
-        }
-        if (workersFile != null) {
-            IOUtil.writeAsLines(new File(workersFile), workers);
-        }
+
+    if (exclusionsFile != null) {
+      IOUtil.writeAsLines(new File(exclusionsFile), exclusions.getServers());
     }
-    
-    private void clearLinks() throws IOException, KeeperException {
-        NamespaceLinksZK    nlzk;
-        
-        nlzk = new NamespaceLinksZK(mc);
-        nlzk.clearAllZK();
+    if (workersFile != null) {
+      IOUtil.writeAsLines(new File(workersFile), workers);
     }
+  }
 
-    private Map<String,String> getHostGroupToClassVarsMap() {
-        return dhtConfig.getHostGroupToClassVarsMap();
+  private void clearLinks() throws IOException, KeeperException {
+    NamespaceLinksZK nlzk;
+
+    nlzk = new NamespaceLinksZK(mc);
+    nlzk.clearAllZK();
+  }
+
+  private Map<String, String> getHostGroupToClassVarsMap() {
+    return dhtConfig.getHostGroupToClassVarsMap();
+  }
+
+  private void displayHostGroupToClassVarsMap() {
+    Map<String, String> hostGroupToClassVars = getHostGroupToClassVarsMap();
+    for (String hostGroup : hostGroupToClassVars.keySet()) {
+      displayForFiltering(hostGroup + "=" + hostGroupToClassVars.get(hostGroup));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.